.circleci/config.yml

```diff
@@ -1,64 +1,129 @@
 version: 2
 jobs:
-  build: # runs not using Workflows must have a `build` job as entry point
-    working_directory: ~/spark-starter-kit # directory where steps will run
-    docker: # run the steps with Docker
-      - image: circleci/ruby:2.5.1-node-browsers # ...with this image as the primary container; this is where all `steps` will run
-        environment: # environment variables for primary container
-          PGHOST: localhost
-          PGUSER: postgres
-          BUNDLE_JOBS: 4
-          BUNDLE_RETRY: 3
-          RAILS_ENV: test
-      - image: circleci/postgres:10.0-alpine
-        environment:
-          POSTGRES_USER: postgres
-          POSTGRES_DB: spark-starter-kit_test
-    steps: # a collection of executable commands
-      - checkout # special step to check out source code to working directory
-
-      - run:
-          name: Which bundler?
-          command: bundle -v
-
-      # bundle install
+  install_ruby_dependencies:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
+    working_directory: ~/spark-starter-kit
+    steps:
+      - checkout
       - restore_cache:
           keys:
-            - spark-starter-kit-{{ checksum "Gemfile.lock" }}
-
+            - ruby-dependencies-v6-{{ checksum "Gemfile.lock" }}
       - run:
-          name: Bundle Install
           command: |
             bundle check || bundle install --path vendor/bundle --deployment
-
       - save_cache:
-          key: spark-starter-kit-{{ checksum "Gemfile.lock" }}
           paths:
             - vendor/bundle
+          key: ruby-dependencies-v6-{{ checksum "Gemfile.lock" }}
 
-      # yarn install
+  install_javascript_dependencies:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
+    working_directory: ~/spark-starter-kit
+    steps:
+      - checkout
       - restore_cache:
           keys:
-            - javascript-cache-v1-{{ checksum "package.json" }}
-
+            - javascript-dependencies-v1-{{ checksum "yarn.lock" }}
       - run:
           command: yarn install --ignore-engines
```

Comment:
let's make sure it doesn't overwrite yarn-lock

Reply:
when there will be any node or yarn version difference between circle's docker image and local environment it will update yarn.lock and it won't match cache key

Reply:
@GrzegorzStochel from what I understand it's not a very good idea to install packages without reflecting it in the package.json, so I would rather leave it as it is (also that's why yarn has no such functionality, I think).

Reply:
i mean't to not overwrite yarn.lock, not package.json

Reply:
while installing rails gems we use --deployment, i'm not sure how yarn works